File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/TrainingRequest.cs
using HRManagementSystem.Domain.Enums;

namespace HRManagementSystem.Domain.Entities;
public class TrainingRequest : BaseEntity
{
    public int EmployeeId { get; set; }
    public Employee? Employee { get; set; }

    public int TrainingCourseId { get; set; }
    public TrainingCourse? TrainingCourse { get; set; }

    public TrainingRequestStatus Status { get; set; } = TrainingRequestStatus.Pending;

    public DateTime RequestDate { get; set; } = DateTime.UtcNow;

    public int? ReviewedByManagerId { get; set; }
    public Employee? Reviewer { get; set; }

    public DateTime? ReviewedAt { get; set; }
    public string? ManagerNote { get; set; }
    public string? EmployeeNote { get; set; }

}

File: frontend/src/app/models/training/training-request.ts
export interface TrainingRequest {
  id: number;
  employeeId: number;
  employeeName: string;
  trainingCourseId: number;
  courseTitle: string;
  status: string | TrainingRequestStatus; // Allow both
  requestDate: string;
  reviewedByManagerId?: number;
  managerName?: string;
  reviewedAt?: string;
  managerNote?: string;
  employeeNote?: string;
}

export enum TrainingRequestStatus {
  Pending = 0,
  Approved = 1,
  Rejected = 2,
}

export interface CreateTrainingRequestDto {
  employeeId: number;
  trainingCourseId: number;
  employeeNote?: string;
}